src/test/java/uk/nickbdyer/httpserver/controllers/FileControllerTest.java

```diff
@@ -50,4 +55,32 @@ public void willReturnFileContentsInBody() throws IOException {
         assertEquals("hello", new String(response.getBody()));
     }
 
+    @Test
+    public void willReturn204ForValidPatchEtag() throws IOException {
```

Comment:
I think that in this case the test name is wrong. I believe I should have a test for a 204 response in this controller, but perhaps I should return 204 regardless of the presence of a "If-Match" header. The test for sha1 matching I expect will be in my "FileManipulation" class, or whatever it gets called when I extract it out.